### PR TITLE
Respect numlock setting

### LINKED ISSUE #337: New greeter ignoring num-lock state conf
Hi, in **/etc/lightdm/io.elementary.greeter.conf** I have uncommented line with **activate-numlock=true**, which worked fine till latest release. Now it is broken and Num-Lock is always off and must be switched manually. Can you fix it, please?

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/79610532-new-greeter-ignoring-num-lock-state-conf?utm_campaign=plugin&utm_content=tracker%2F60068889&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F60068889&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by davidmhewitt**
Fixes #337

### CODE REVIEW

**Comment by jeremypw** [comment]
This does not seem to work for me - numlock is always off in greeter still.  I am not sure how to debug this however.

**Comment by m93a** [comment]
@jeremypw The PR is not merged yet, are you sure you tested the correct branch on your system?

**Comment by jeremypw** [comment]
Further investigation indicates that `numlockx on` is being correctly spawned and that no errors are returned or output from this process.  However, event when called in several places in the greeter, no change in the numlock status actually occurs.  Not sure why :-(

**Comment by jeremypw** [comment]
Even weirder, the CapsLockRevealer is receiving the correct signal and setting reveal to `true` with the appropriate message.  Nevertheless it is not visible on screen.

File: src/MainWindow.vala [hunk 1]
```diff
@@ -293,6 +293,14 @@ public class Greeter.MainWindow : Gtk.ApplicationWindow {
...
 
         maximize_window ();
+
+        if (settings.activate_numlock) {
+            try {
+                Process.spawn_async (null, { "numlockx", "on" }, null, SpawnFlags.SEARCH_PATH, null, null);
+            } catch (Error e) {
+                warning ("Unable to spawn numlockx to set numlock state");
+            }
+        }
```
[the remark this answers is not in the document]
Comment (davidmhewitt) [reply src/MainWindow.vala]: So the problem is not that numlock isn't activated, it's that the visual indicator in the UI does not update properly?

[the remark this answers is not in the document]
Comment (jeremypw) [reply src/MainWindow.vala]: Yes, it seems so.

[the remark this answers is not in the document]
Comment (davidmhewitt) [reply src/MainWindow.vala]: Resolved in the latest commit

**Comment by jeremypw** [comment]
Yup, that fixes it too.  It seems a new Window is constructed each time the greeter is shown as far as I can tell.

**Pull request merged**